hphp/runtime/vm/jit/unique-stubs.cpp

```diff
@@ -1034,12 +1034,15 @@ TCA emitDecRefGeneric(CodeBlock& cb, DataBlock& data) {
   CGMeta meta;
 
   auto const start = vwrap(cb, data, meta, [] (Vout& v) {
+    v << vregrestrict{};
     v << stublogue{};
 
     auto const rdata = rarg(0);
     auto const rtype = rarg(1);
 
     auto const destroy = [&] (Vout& v) {
+      v << vregallow{};
```

Comment:
Initially I wanted to do the same thing, but it's difficult for the PhysRegSaver to know if it's currently within a restricted region in order to emit the `vregstrict` after the registers are restored. This information isn't done until the analysis to compute the dominating vreg restictions is performed while lower.